src/pip/_internal/utils/logging.py

```diff
@@ -162,7 +162,11 @@ def emit(self, record: logging.LogRecord) -> None:
         else:
             message = self.format(record)
             renderable = self.render_message(record, message)
-            if record.levelno is not None:
+            # If a custom color is passed use it, otherwise use default colors.
+            color_attribute = "color"
+            if hasattr(record, color_attribute):
+                style = Style(color=getattr(record, color_attribute))
```

Comment:
It's usually better to follow EAFP in Python. As in, access the attribute with a dot-syntax and not a dynamic function call. Could go for try/except in this case.

Reply:
I find that this is clearer.

Reply:
Wouldn't it confuse things like mypy?

Reply:
Actually mypy will be confused by dot-syntax because `record` class doesn't have a `color` attribute by default so mypy screams if you try to access `color` using `record.color`.
FWIW, this is how `color` or any other extra attribute are assigned to the record:
```
        if extra is not None:
            for key in extra:
                if (key in ["message", "asctime"]) or (key in rv.__dict__):
                    raise KeyError("Attempt to overwrite %r in LogRecord" % key)
                rv.__dict__[key] = extra[key]
```
https://github.com/python/cpython/blob/3.10/Lib/logging/__init__.py#L1593:L1597